src/zebra_puzzles/clue_removal.py

```diff
@@ -251,5 +257,75 @@ def remove_redundant_clues_with_solver(
         if completeness == 1:
             del chosen_clues[i]
             del chosen_constraints[i]
+            del chosen_clue_types[i]
+
+    return chosen_clues, chosen_constraints, chosen_clue_types
+
+
+def remove_red_herrings(
+    prompt: str,
+    i_red_herrings: str,
```

Comment:
I'd rename this to something like `red_herring_indices_str`, since `indices` is more descriptive than `i`, and the `_str` emphasises that this is not just the indices, it's the indices packed into a string.